examples/visionProcessing/ageDetection/ageDetection.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2024 The IceFlow Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "iceflow/consumer.hpp"
+#include "iceflow/dag-parser.hpp"
+#include "iceflow/iceflow.hpp"
+#include "iceflow/measurements.hpp"
+#include "iceflow/producer.hpp"
+#include "iceflow/serde.hpp"
+
+#include <ndn-cxx/face.hpp>
+#include <ndn-cxx/util/logger.hpp>
+
+#include <csignal>
+#include <functional>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/opencv.hpp>
+
+NDN_LOG_INIT(iceflow.examples.visionprocessing.ageDetection);
+
+iceflow::Measurement *measurementHandler;
+
+void signalCallbackHandler(int signum) {
+  measurementHandler->recordToFile();
+  exit(signum);
+}
+
+class AgeDetection {
+public:
+  AgeDetection(const std::string &protobufFile, const std::string &mlModel)
+      : ageNet(cv::dnn::readNet(mlModel, protobufFile)) {}
+
+  void ageDetection(std::vector<uint8_t> encodedCropped,
+                    std::function<void(std::vector<uint8_t>)> push) {
```

Comment:
Maybe this method could rather be renamed into something `detectAge`, since method names should generally use verbs instead of nouns.